modeltrans/forms.py

```diff
@@ -0,0 +1,380 @@
+        - The list order determines the order of fields in the form.
+        - included languages can also be passed via form kwargs to customize on the fly, useful for
+          generating translation forms for a specific language.
+    - fallback_language: for adapting the fallback language specification on the fly and override the Meta option and/or
+    model translation field custom fallback.
+    # TODO FUTURE - fallback_readonly: boolean defining whether fallback fields are editable or only for reference.
+    #  This is useful if editing fallback languages is to be restricted. Can be passed as kwarg too.
+    Form parameters take priority of Meta options.
+
+    For the fallback language the following priority holds:
+    1) fallback_language passed as form parameter: Form(fallback_language="fr")
+    2) the Meta option "fallback_language":
+        e.g. Meta:
+                fallback_language = "fr"
+    3) a custom fallback of a model instance set via "fallback_language_field":
+        e.g. i18n = TranslationField(fields=("title", "header"), fallback_language_field="language_code")
+    4) The default language of the system. If not Meta option is given fallback reverts to get_default_language()
+
+    Code example:
+        class ChallengeModel(models.Model):
+            language_code = CharField()
+            title = CharField()
+            description = CharField()
+            i18n = TranslationField(fields=("title", "description"), fallback_language_field="language_code")
+
+
+        class ChallengeModelForm(RefactoredModelTransForm):
+
+            class Meta:
+                fields = ["title", "description"] or exclude = ["language_code"]
+                included_languages = ["browser", "es", "fallback"]
+                # TODO FUTURE later fallback_readonly = False
+
+
+        class ChallengeCreateUpdateView(GenericCreateUpdateView):
+            '''An update view with a browser language, spanish and browser field for translations.'''
+            model = Challenge
+            form_class = ChallengeModelForm()
+
+
+        class ChallengeTranslationView(GenericCreateUpdateView):
+            '''
+            A translation view with a specific translation language field and non-editable fallback field as reference.
+            '''
+            model = Challenge
+            form_class = ChallengeModelForm(
+                included_languages=[translation_language_code, "fallback"],
+                # TODO FUTURE fallback_readonly=True
+            )
+    """
+
+    # TODO FUTURE fallback_readonly=None
+    def __init__(self, *args, included_languages=None, fallback_language=None, **kwargs):
+        """Prune the translation fields based on included languages and fallback_language."""
+
+        self.model_i18n_field = get_i18n_field(self._meta.model)
+        self.included_languages = included_languages or self._meta.included_languages
+        self.form_i18n_fields = self.form_i18n_fields()
```

Comment:
```suggestion
        self.form_i18n_fields = [field for field in self.model_i18n_field.fields if field in self.base_fields.keys()]
```

Do we need the `form_`-prefix here?